mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -740,7 +740,7 @@ private[ml] object RandomForest extends Logging {
               val categoryStats =
                 binAggregates.getImpurityCalculator(nodeFeatureOffset, featureValue)
               val centroid = if (categoryStats.count != 0) {
-                categoryStats.predict
+                categoryStats.prob(categoryStats.predict)
```

Comment:
I don't believe this is correct. Ordering by the probability of the prediction is essentially the same as ordering by impurity. That's because when the impurity is low, the predicted value will have high probability and vice versa. 

From Hastie, Tibshirani, and Friedman:
"We order the predictor classes according to the proportion falling in outcome class 1. Then we split this predictor as if it were an ordered predictor."

For binary category I think it should be as @jkbradley suggested `categoryStats.stats(1)`